cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -241,4 +241,15 @@ public Secret cruiseControlSecret() {
             isMaintenanceTimeWindowsSatisfied);
     }
 
+    @Override
+    protected boolean isCaCertThumbprintChanged() {
+        if (zkNodesSecret != null) {
+            String clusterCaThumbprintAnno = zkNodesSecret.getMetadata().getAnnotations().get("cluster-ca-thumbprint");
```

Comment:
Just curious...the method invocation of `isCaCertThumbprintChanged` cannot produce here NPE? (there will always be in annotations `cluster-ca-thumbprint` when the method is invoked?)

Reply:
Yeah when the Secret is created, it always has the annotation.

Reply:
And what about upgrade?

Reply:
Damn ... good point, coming from an older release of the Strimzi operator! I will check this, thanks guys.